service-map/src/main/java/fi/nls/oskari/map/data/domain/GFIRequestParams.java

```diff
@@ -158,5 +159,26 @@ private String getBaseQueryURL() {
         }
         return queryUrl;
     }
+    private String getInfoFormat () {
+        String infoFormat = layer.getGfiType();
+        if (infoFormat != null && !infoFormat.isEmpty()) {
+            return infoFormat;
+        }
+        JSONObject formats = layer.getCapabilities().optJSONObject(KEY_FORMATS);
+        if (formats == null) {
+            return DEFAULT_FORMAT;
+        }
+        String value = formats.optString(KEY_VALUE);
+        if (!value.isEmpty()) {
+            return value;
+        }
+        List <String> available = JSONHelper.getArrayAsList(JSONHelper.getJSONArray(formats, KEY_AVAILABLE));
+        for(String format : SUPPORTED_GET_FEATURE_INFO_FORMATS) {
+            if (available.contains(format)) {
+                return format;
+            }
+        }
+        return DEFAULT_FORMAT;
```

Comment:
I'm wondering if we might as well cancel the request if any of the available are not supported or should we fallback to using the first from available and hope for the best?